webapp/tests/karma/ts/services/message-contact.service.spec.ts

```diff
@@ -0,0 +1,292 @@
+import { GetDataRecordsService } from '@mm-services/get-data-records.service';
+import { DbService } from '@mm-services/db.service';
+import { AddReadStatusService } from '@mm-services/add-read-status.service';
+
+describe('Message Contacts Service', () => {
+  let service: MessageContactService;
+  let dbService;
+  let getDataRecordsService;
+  let hydrateMessagesService;
+
+  beforeEach(() => {
+    const dbMock = { get: () => ({ query: sinon.stub() }) };
+    const getDataRecordsMock = { get: sinon.stub() };
+    const hydrateMessagesMock = { hydrate: sinon.stub() };
+
+    TestBed.configureTestingModule({
+      providers: [
+        { provide: HydrateMessagesService, useValue: hydrateMessagesMock },
+        { provide: GetDataRecordsService, useValue: getDataRecordsMock },
+        { provide: DbService, useValue: dbMock },
+        AddReadStatusService
+      ]
+    });
+
+    service = TestBed.inject(MessageContactService);
+    hydrateMessagesService = TestBed.inject(HydrateMessagesService);
+    getDataRecordsService = TestBed.inject(GetDataRecordsService);
+    dbService = TestBed.inject(DbService);
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it('should return minimum limit', () => {
+    expect(service.minLimit).to.equal(50);
+  });
+
+  describe('getList()', () => {
+    it('should build list', () => {
+      const dbRows = {
+        rows: [
+          { id: 'some_id1', value: { id: 'id1' } },
+          { id: 'some_id2', value: { id: 'id2' } },
+          { id: 'some_id3', value: { id: 'id3' } },
+          { id: 'some_id4', value: { id: 'id4' } },
+        ]
+      };
+      dbService.get = () => ({
+        query: sinon.stub().resolves(dbRows),
+        allDocs: sinon.stub().resolves(dbRows)
+      });
+      getDataRecordsService.get.resolves([
+        { _id: 'id1' },
+        { _id: 'id2' },
+        { _id: 'id3' },
+        { _id: 'id4' },
+      ]);
+      hydrateMessagesService.hydrate.resolves([
+        { id: 'some_id1', hydrated: true },
+        { id: 'some_id2', hydrated: true },
+        { id: 'some_id3', hydrated: true },
+        { id: 'some_id4', hydrated: true },
+      ]);
+
+      return service
+        .getList()
+        .then(list => {
+          expect(getDataRecordsService.get.callCount).to.equal(1);
+          expect(getDataRecordsService.get.args[0]).to.deep.equal([['id1', 'id2', 'id3', 'id4'], { include_docs: true }]);
+          expect(hydrateMessagesService.hydrate.callCount).to.equal(1);
+          expect(hydrateMessagesService.hydrate.args[0]).to.deep.equal([[
+            { id: 'some_id1', value: { id: 'id1' }, doc: { _id: 'id1' } },
+            { id: 'some_id2', value: { id: 'id2' }, doc: { _id: 'id2' } },
+            { id: 'some_id3', value: { id: 'id3' }, doc: { _id: 'id3' } },
+            { id: 'some_id4', value: { id: 'id4' }, doc: { _id: 'id4' } },
+          ]]);
+          expect(list).to.deep.equal([
+            { id: 'some_id1', hydrated: true, read: true },
```

Comment:
So, it's read status incorrect?

Reply:
Looking at the code that sets this value in AddReadStatusService, appears to be expected 🤔
models.forEach((model, i) => {
    const row = response.rows[i];
    model.read = !!(row.value && !row.value.deleted); // doc exists.
});

Reply:
It's a bit concerning having one array to access another by the same positioning idx, looks like it's working, but could easily break. Maybe something to improve in another task 🤔

Reply:
Yea, but you're using the same mock for the allDocs request that you're using for getting messages to getting read docs. 
We should either mock the meta db alldocs separately or mock the read docs service.

Reply:
Ah because both services where using the same mocked data of DB Service 👍. So, now I'm mocking AddReadStatus Service as it was originally, it returns what it receives.